src/store/tokens-metadata/tokens-metadata-reducers.ts

```diff
@@ -1,110 +1,97 @@
 import { createReducer } from '@reduxjs/toolkit';
+import { persistReducer } from 'redux-persist';
 
 import { emptyTokenMetadata } from 'src/token/interfaces/token-metadata.interface';
-import { getTokenSlug } from 'src/token/utils/token.utils';
-import { isDefined } from 'src/utils/is-defined';
+import { getTokenSlug, toTokenSlug } from 'src/token/utils/token.utils';
+import { transformWhitelistToTokenMetadata } from 'src/utils/token-metadata.utils';
 
 import { createEntity } from '../create-entity';
-import { setNewTokensMetadata } from '../migration/migration-actions';
+import { AsyncMmkvStorage } from '../mmkv-storage';
 
 import {
   addKnownSvg,
-  addTokensMetadataAction,
+  putTokensMetadataAction,
   loadTokenSuggestionActions,
   loadTokensMetadataActions,
   loadWhitelistAction,
   removeKnownSvg
 } from './tokens-metadata-actions';
 import { tokensMetadataInitialState, TokensMetadataState } from './tokens-metadata-state';
 
-export const tokensMetadataReducers = createReducer<TokensMetadataState>(tokensMetadataInitialState, builder => {
+const tokensMetadataReducers = createReducer<TokensMetadataState>(tokensMetadataInitialState, builder => {
   builder.addCase(loadTokensMetadataActions.submit, state => {
     state.isLoading = true;
   });
-  builder.addCase(loadTokensMetadataActions.success, state => {
+
+  builder.addCase(loadTokensMetadataActions.success, (state, { payload }) => {
     state.isLoading = false;
+
+    for (const metadata of payload) {
```

Comment:
`payload.forEach` should work even faster. Please pay attention to other places where `for (... of ...)` is used